### PR TITLE
Add hooks

### PR DESCRIPTION
**Pull request by vderunov**

### CODE REVIEW

File: src/useDelegateCollateral.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+}) {
+  const { chainId, queryClient } = useSynthetix();
+  const errorParser = useErrorParser();
+
+  const accountId = useSelectedAccountId({
+    accountId: accountIdFromParams,
+    provider,
+    walletAddress,
+  });
+  const poolId = useSelectedPoolId({ poolId: poolIdFromParams });
```
Comment (noisekit) [inline src/useDelegateCollateral.ts]: these should be in the App and not SDK as they are related to URL params. useDelegateCollateral should take `accountId` and `poolId` arguments

Comment (noisekit) [reply src/useDelegateCollateral.ts]: "convenience" hooks `useSelectedPoolId` and `useSelectedAccountId` could remain here in SDK though.

Comment (vderunov) [reply src/useDelegateCollateral.ts]: Removed: `const poolId = useSelectedPoolId({ poolId: poolIdFromParams });`

Updated hook parameters:

```javascript
export function useDelegateCollateral({
  provider,
  walletAddress,
  collateralTypeTokenAddress,
  poolId,
  accountId,
  onSuccess,
}: {
```

Comment (noisekit) [reply src/useDelegateCollateral.ts]: great

**Review by noisekit — APPROVED** [review]

**Pull request merged**